### PR TITLE
Add Correct Email Messaging References

### PR DESCRIPTION
**Pull request by michaelwehrley**
* Add spec to check for correctly referenced url.
* Change tool reference from 'MESSAGES' to tool title.
* Update fixtures to tests tool title with a space.

[Finished: #105498898]

### CODE REVIEW

File: app/mailers/think_feel_do_engine/message_notifications.rb [hunk 1]
```diff
@@ -8,6 +8,12 @@ def new_for_coach(coach, group)
     end
 
     def new_for_participant(participant)
+      @context_name = participant
+                      .current_group
+                      .arm
+                      .bit_core_tools
+                      .find_by_title("MESSAGES")
```
Comment (ericcf) [inline app/mailers/think_feel_do_engine/message_notifications.rb]: Won't it be very brittle to search by a hard-coded database value that could change?

Comment (ericcf) [inline app/mailers/think_feel_do_engine/message_notifications.rb]: Maybe better to search by the `type` field?

File: db/migrate/20151102160119_add_slug_to_bit_core_tools.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+class AddSlugToBitCoreTools < ActiveRecord::Migration
+  def up
+    add_column :bit_core_tools, :slug, :string
```
Comment (ericcf) [inline db/migrate/20151102160119_add_slug_to_bit_core_tools.rb]: Won't it be more difficult to write logic that uses this field if it's allowed to be null?

File: app/mailers/think_feel_do_engine/message_notifications.rb (hunk 1 above)
Comment (michaelwehrley) [inline app/mailers/think_feel_do_engine/message_notifications.rb]: Good point.  I forgot about the type field.  Thanks!

File: db/migrate/20151102160119_add_slug_to_bit_core_tools.rb (hunk 2 above)
Comment (ericcf) [inline db/migrate/20151102160119_add_slug_to_bit_core_tools.rb]: Sorry, just saw line 11!

Comment (michaelwehrley) [inline db/migrate/20151102160119_add_slug_to_bit_core_tools.rb]: I think there is a different solution that may be better.  I will update with your comment and resubmit.  Thanks!

**Comment by michaelwehrley** [comment]
@ericcf The url wasn't even the problem.  I had to fix the url reference which I did, and then stupidly got caught up in the slug idea.  I already did test out on staging that `https://sunnyside-staging.cbits.northwestern.edu/navigator/contexts/Contact%20Moderator` works.  No problem.

**Comment by ericcf** [comment]
Thanks!

**Pull request merged**